pkg/feature/servicemesh/servicemesh_feature.go

```diff
@@ -0,0 +1,59 @@
+package servicemesh
+
+import (
+	"path"
+	"path/filepath"
+
+	ctrlLog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+)
+
+const templatesDir = "templates/servicemesh"
+
+var log = ctrlLog.Log.WithName("features")
+
+func ConfigureServiceMeshFeatures(s *feature.FeaturesInitializer) error {
+	var rootDir = filepath.Join(feature.BaseOutputDir, s.DSCInitializationSpec.ApplicationsNamespace)
+	if err := feature.CopyEmbeddedFiles(templatesDir, rootDir); err != nil {
+		return err
+	}
+
+	serviceMeshSpec := s.ServiceMesh
+
+	smcpCreation, errSmcp := feature.CreateFeature("control-plane-creation").
```

Comment:
can we rename it here to be more specific than `control-plane-creation`

Reply:
Do you have any particular name in mind? Naming is hard and I'm out of ideas.

Reply:
maybe something specific for servicemesh? `control-plane-creation` is bit to general for me

Reply:
I prefixed everything in this file with `mesh-` so that we can have more context in the logs.